logs_api/lib/opentelemetry/internal/proxy_logger_provider.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Internal
+    # @api private
+    #
+    # {ProxyLoggerProvider} is an implementation of {OpenTelemetry::Logs::LoggerProvider}.
+    # It is the default global logger provider returned by OpenTelemetry.logger_provider.
+    # It delegates to a "real" LoggerProvider after the global logger provider is registered.
+    # It returns {ProxyLogger} instances until the delegate is installed.
+    class ProxyLoggerProvider < Logs::LoggerProvider
+      Key = Struct.new(:name, :version)
+      private_constant(:Key)
+      # Returns a new {ProxyLoggerProvider} instance.
+      #
+      # @return [ProxyLoggerProvider]
+      def initialize
+        super
+
+        @mutex = Mutex.new
+        @registry = {}
+        @delegate = nil
+      end
```

Comment:
We want the instrumentation registry merged first so these features aren't random